osu.Game/Rulesets/Objects/Types/ComboIndex.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+
+namespace osu.Game.Rulesets.Objects.Types
+{
+    /// <summary>
+    /// Represents a manual and automated combo indices.
+    /// </summary>
+    public readonly struct ComboIndex : IEquatable<ComboIndex>
+    {
+        /// <summary>
+        /// The ordinal index value of the combo in relation to the beatmap. Ignores <see cref="IHasCombo.ComboOffset"/>.
+        /// </summary>
+        /// <remarks>
+        /// This is separated from <see cref="WithOffset"/> to disallow
+        /// manual combo colouring when not on the right skin to apply.
+        /// </remarks>
+        public readonly int Ordinal;
```

Comment:
so is this the combo's index in the beatmap? if so, you should also look at `IHasCombo.ComboOffset` as it may be similar to what is provided here.

as it stands, this documentation/class feels very hard to understand.

Reply:
isn't `IHasCombo.ComboOffset` just the amount of combos to skip from current combo index in the processing? this index doesn't let that in its calculation: `Ordinal = lastObj.Ordinal + 1;` (which just increments by 1 per new combo, or as you said "the combo's index in the beatmap")

If osu!stable provides something much saner / straightforward than this, would be better to let me know I suppose.